src/geodesic.rs

```diff
@@ -2693,15 +2690,21 @@ mod tests {
 
         geodtest_basic(
             test_input_path(),
-            |_line_num, &(lat1, lon1, azi1, lat2, lon2, azi2, s12, a12, m12, S12)| {
+            |line_num, &(lat1, lon1, azi1, lat2, lon2, azi2, s12, a12, m12, S12)| {
                 let g = g.lock().unwrap();
                 let (s12_out, azi1_out, azi2_out, m12_out, _M12_out, _M21_out, S12_out, a12_out) =
                     g.inverse(lat1, lon1, lat2, lon2);
                 assert_approx_eq!(s12, s12_out, 8e-9);
                 assert_approx_eq!(azi1, azi1_out, 2e-2);
                 assert_approx_eq!(azi2, azi2_out, 2e-2);
                 assert_approx_eq!(m12, m12_out, 5e-5);
-                assert_approx_eq!(S12, S12_out, 3e10); // Note: unreasonable tolerance
+                // Our area calculation differs significantly (~1e7) from the value in GeodTest.dat for
+                // line 400001, BUT our value also perfectly matches the value returned by GeographicLib
+                // (C++) 1.51. Here's the problem line, for reference:
+                // 4.199535552987 0 90 -4.199535552987 179.398106343454992238 90 19970505.608097404994 180 0 
+                if line_num != 400001 {
+                    assert_approx_eq!(S12, S12_out, 3e10); // Note: unreasonable tolerance
```

Comment:
My understanding was that the reason we had an unreasonable tolerance was because case 400001 was behaving badly.

Now that we're skipping 400001 do we still need the unreasonable tolerance?

Reply:
Yes, I forgot to update that before creating the pull request. The answer is that we can now use a smaller tolerance, but not exactly reasonable. If I remember correctly, the bug fixes I applied should allow a tolerance of 2e7 instead of 3e10 for line 400001 (if we didn't exempt it). I can't remember how the other problem cases compared. I'll try to update that later this evening.

Reply:
Looking back at this, I see that I was using a relative error to pick a "worst" line for S12, and it basically just picked the first one that had expected 0 and actual with a large value, which all end up with a relative error of 2. There are more with relative errors of 2, and some of those still have absolute errors in the same range as line 400001, so it's not clear that we can reduce the tolerance at all yet. Just to be safe, I plan to revise my "worst line" metrics to pick a separate worst line for absolute and relative error for any value where I think we care about relative error (I still think it makes sense for S12), and rerun with the updated metrics using code from both before and after the changes in this PR, to make sure that everything is really moving in the right direction. I don't expect to finish that review tonight.

Reply:
ok - so my understanding is you'd like to make some further changes to this tolerance checking before merging - is that right?